kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusKieServerExtension.java

```diff
@@ -59,13 +72,62 @@ public boolean isInitialized() {
 
     @Override
     public boolean isActive() {
-        return !disabled;
+        return disabled == false;
     }
 
     @Override
     public void init(KieServerImpl kieServer, KieServerRegistry registry) {
         this.context = registry;
+
+        registerDefaultDescriptor();
+
+        //Prometheus Monitoring
+        KieServerExtension jBPMExtension = context.getServerExtension(JbpmKieServerExtension.EXTENSION_NAME);
+        if (jBPMExtension != null) {
+
+            final KModuleDeploymentService deploymentService = jBPMExtension.getAppComponents(KModuleDeploymentService.class);
+            if (deploymentService != null) {
+                deploymentService.addListener(new PrometheusDeploymentEventListener());
+            }
+
+            final ExecutorServiceImpl executorService = jBPMExtension.getAppComponents(ExecutorServiceImpl.class);
+            if (executorService != null) {
+                executorService.addAsyncJobListener(new PrometheusJobListener());
+            }
+
+            final QueryServiceImpl queryService = jBPMExtension.getAppComponents(QueryServiceImpl.class);
+            if (queryService != null) {
+                final DataSetDefRegistry dataSetDefRegistry = queryService.getDataSetDefRegistry();
+                final PrometheusDataSetListener listener = new PrometheusDataSetListener(dataSetDefRegistry);
+                listener.init();
+                dataSetDefRegistry.addListener(listener);
```

Comment:
Nitpicking: is this bi-directional relationship really needed?

Reply:
its because I'm using the _PrometheusDataSetListener_ as a series of different listeners related to data set API: _DataSetDefRegistryListener_, _DataSetPreprocessor_,                                            _DataSetPostProcessor_ so not only it is a listener for the registry but also for every time a dataset is added or modified, we need to register the pre and post processor again.

Reply:
+1